src/libsync/propagateuploadng.cpp

```diff
@@ -30,23 +30,93 @@
 #include <QNetworkAccessManager>
 #include <QFileInfo>
 #include <QDir>
+
 #include <cmath>
 #include <cstring>
 
 namespace OCC {
 
-QUrl PropagateUploadFileNG::chunkUrl(int chunk)
+QUrl PropagateUploadFileNG::chunkUrl(qint64 chunk)
 {
     QString path = QLatin1String("remote.php/dav/uploads/")
         + propagator()->account()->davUser()
         + QLatin1Char('/') + QString::number(_transferId);
     if (chunk >= 0) {
-        // We need to do add leading 0 because the server orders the chunk alphabetically
-        path += QLatin1Char('/') + QString::number(chunk).rightJustified(8, '0');
```

Comment:
Major: Why don't you have to add leading zeros anymore? Even if the server got changed, we need to be backwards-compatible!

Reply:
didn't think about backward compatibility, how far back does this bug go? I saw that the server orders them properly now. But `rightJustified(8)` won't be quite right for me, since offsets will need a larger space than that ... How high should I go??

Reply:
Since zsync uploads are gated behind a capability that'll only be enabled for servers that support the right sorting you don't need to worry about that. But non-zsync upload must continue to work with servers that use alphabetic sort there. (so pad non-zsync chunk ids?)

Reply:
But even for non-zsync uploads I am now using the chunkId set as the offset, so `rightJustified(8)` isn't going to cover really large files. It would have in the previous incremental value of chunkId. Might need to be `32` or larger 😟 .